### ISSUE TITLE
Gallery crashing on my Nexus 5X

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#3177]
```
I/flutter : Observatory listening on http://127.0.0.1:41310
I/flutter : Diagnostic server listening on http://127.0.0.1:41393
W/ActivityManager: Scheduling restart of crashed service com.qualcomm.qcrilmsgtunnel/.QcrilMsgTunnelService in 1000ms
W/ActivityManager: Scheduling restart of crashed service com.qualcomm.atfwd/.AtFwdService in 10890ms
W/System.err: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
W/System.err: 	at org.domokit.sky.shell.FlutterSemanticsToAndroidAccessibilityBridge.removePersistentNode(FlutterSemanticsToAndroidAccessibilityBridge.java:261)
W/System.err: 	at org.domokit.sky.shell.FlutterSemanticsToAndroidAccessibilityBridge$PersistentAccessibilityNode.update(FlutterSemanticsToAndroidAccessibilityBridge.java:326)
W/System.err: 	at org.domokit.sky.shell.FlutterSemanticsToAndroidAccessibilityBridge.updateSemanticsNode(FlutterSemanticsToAndroidAccessibilityBridge.java:241)
W/System.err: 	at org.domokit.sky.shell.FlutterSemanticsToAndroidAccessibilityBridge.updateSemanticsTree(FlutterSemanticsToAndroidAccessibilityBridge.java:219)
W/System.err: 	at org.chromium.mojom.semantics.SemanticsListener_Internal$Stub.accept(SemanticsListener_Internal.java:79)
W/System.err: 	at org.chromium.mojo.bindings.RouterImpl.handleIncomingMessage(RouterImpl.java:236)
W/System.err: 	at org.chromium.mojo.bindings.RouterImpl.access$000(RouterImpl.java:18)
W/System.err: 	at org.chromium.mojo.bindings.RouterImpl$HandleIncomingMessageThunk.accept(RouterImpl.java:30)
W/System.err: 	at org.chromium.mojo.bindings.Connector.readAndDispatchMessage(Connector.java:245)
W/System.err: 	at org.chromium.mojo.bindings.Connector.readOutstandingMessages(Connector.java:204)
W/System.err: 	at org.chromium.mojo.bindings.Connector.onAsyncWaiterResult(Connector.java:170)
W/System.err: 	at org.chromium.mojo.bindings.Connector.access$100(Connector.java:25)
W/System.err: 	at org.chromium.mojo.bindings.Connector$AsyncWaiterCallback.onResult(Connector.java:150)
W/System.err: 	at org.chromium.mojo.system.impl.CoreImpl.onAsyncWaitResult(CoreImpl.java:573)
W/System.err: 	at org.chromium.base.SystemMessageHandler.nativeDoRunLoopOnce(Native Method)
W/System.err: 	at org.chromium.base.SystemMessageHandler.handleMessage(SystemMessageHandler.java:53)
W/System.err: 	at android.os.Handler.dispatchMessage(Handler.java:102)
W/System.err: 	at android.os.Looper.loop(Looper.java:148)
W/System.err: 	at android.app.ActivityThread.main(ActivityThread.java:5422)
W/System.err: 	at java.lang.reflect.Method.invoke(Native Method)
W/System.err: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
W/System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
F/flutter : [FATAL:jni_android.cc(249)] Check failed: false. Please include Java exception stack in crash report
--------- beginning of crash
F/libc    : Fatal signal 6 (SIGABRT), code -6 in tid 28140 (MaterialGallery)
F/DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
F/DEBUG   : Build fingerprint: 'google/bullhead/bullhead:6.0.1/MHC19J/2595691:user/release-keys'
F/DEBUG   : Revision: 'rev_1.0'
F/DEBUG   : ABI: 'arm'
F/DEBUG   : pid: 28140, tid: 28140, name: MaterialGallery  >>> io.flutter.MaterialGallery <<<
F/DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
F/DEBUG   : Abort message: '[FATAL:jni_android.cc(249)] Check failed: false. Please include Java exception stack in crash report
F/DEBUG   : '
F/DEBUG   :     r0 00000000  r1 00006dec  r2 00000006  r3 f716eb7c
F/DEBUG   :     r4 f716eb84  r5 f716eb34  r6 0000000b  r7 0000010c
F/DEBUG   :     r8 ff8b58a0  r9 00000000  sl ff8b5a08  fp f4f7d728
F/DEBUG   :     ip 00000006  sp ff8b5400  lr f6ee0f85  pc f6ee3374  cpsr 400f0010
F/DEBUG   : 
F/DEBUG   : backtrace:
F/DEBUG   :     #00 pc 00042374  /system/lib/libc.so (tgkill+12)
F/DEBUG   :     #01 pc 0003ff81  /system/lib/libc.so (pthread_kill+32)
F/DEBUG   :     #02 pc 0001c73f  /system/lib/libc.so (raise+10)
F/DEBUG   :     #03 pc 000198f1  /system/lib/libc.so (__libc_android_abort+34)
F/DEBUG   :     #04 pc 000174b0  /system/lib/libc.so (abort+4)
F/DEBUG   :     #05 pc 000d3a55  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #06 pc 000c237b  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #07 pc 000b9f53  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #08 pc 0008e413  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #09 pc 000e2f2d  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #10 pc 000e289b  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #11 pc 000d3f3b  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #12 pc 000c3c77  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #13 pc 000c4111  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #14 pc 000c44a1  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #15 pc 000c4eff  /data/app/io.flutter.MaterialGallery-2/lib/arm/libsky_shell.so
F/DEBUG   :     #16 pc 0025b505  /data/app/io.flutter.MaterialGallery-2/oat/arm/base.odex (offset 0x218000)
W/ActivityManager:   Force finishing activity io.flutter.MaterialGallery/org.domokit.sky.shell.SkyActivity
F/DEBUG   : 
F/DEBUG   : Tombstone written to: /data/tombstones/tombstone_03
W/ActivityManager: Exception thrown during pause
W/ActivityManager: android.os.DeadObjectException
W/ActivityManager: 	at android.os.BinderProxy.transactNative(Native Method)
W/ActivityManager: 	at android.os.BinderProxy.transact(Binder.java:503)
W/ActivityManager: 	at android.app.ApplicationThreadProxy.schedulePauseActivity(ApplicationThreadNative.java:727)
W/ActivityManager: 	at com.android.server.am.ActivityStack.startPausingLocked(ActivityStack.java:867)
W/ActivityManager: 	at com.android.server.am.ActivityStack.finishActivityLocked(ActivityStack.java:2907)
W/ActivityManager: 	at com.android.server.am.ActivityStack.finishTopRunningActivityLocked(ActivityStack.java:2763)
W/ActivityManager: 	at com.android.server.am.ActivityStackSupervisor.finishTopRunningActivityLocked(ActivityStackSupervisor.java:2760)
W/ActivityManager: 	at com.android.server.am.ActivityManagerService.handleAppCrashLocked(ActivityManagerService.java:12050)
W/ActivityManager: 	at com.android.server.am.ActivityManagerService.makeAppCrashingLocked(ActivityManagerService.java:11946)
W/ActivityManager: 	at com.android.server.am.ActivityManagerService.crashApplication(ActivityManagerService.java:12635)
W/ActivityManager: 	at com.android.server.am.ActivityManagerService.handleApplicationCrashInner(ActivityManagerService.java:12142)
W/ActivityManager: 	at com.android.server.am.NativeCrashListener$NativeCrashReporter.run(NativeCrashListener.java:86)
```

Looks like Java AX code.

FYI @Hixie @jason-simmons @HansMuller

**Comment by Hixie** [comment]
i can look on Monday, but in the meantime, do you have steps to reproduce?  do you have any a11y things turned on, e.g.? or Google now on tap?

**Comment by eseidelGoogle** [comment]
Sorry, repro steps are to click on "demos" And "components" list headers repeatedly/alternating.  This is my personal (@gmail) phone, I don't think I have AX turned on, but I probably have Now On Tap, etc.

**Comment by Hixie** [comment]
can you repro on any other device? if not, did it repro on other 5Xs? Do you get any dart exceptions in checked mode?

**Comment by Hixie** [comment]
We believe this is caused by apps like DashLane or LastPass which turn on accessibility.
